### PR TITLE
[stable/kube-state-metrics] add optional prometheus service monitor

### PR DESCRIPTION
**Pull request by rmb938**
#### What this PR does / why we need it:

add optional prometheus service monitor to kube-state-metrics

#### Special notes for your reviewer:

other charts like the nginx-ingress and prometheus-node-exporter have this as a feature so I'm adding it to kube-state-metrics as well

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by paulczar** [comment]
pls resolve the merge conflicts.

**Comment by rmb938** [comment]
@paulczar conflicts have been resolved

**Comment by rmb938** [comment]
@paulczar Latest conflicts have been resolved. Could we look to getting this merged sometime soon?

File: stable/kube-state-metrics/Chart.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ keywords:
 - metric
 - monitoring
 - prometheus
-version: 0.16.0
+version: 0.16.1
```
Comment (cpanato) [inline stable/kube-state-metrics/Chart.yaml]: this chart is in the stable folder, lets bump that to 1.0.0

Comment (rmb938) [reply stable/kube-state-metrics/Chart.yaml]: done

**Pull request merged**